lib/internal/loader/ModuleJob.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+const { SafeSet, SafePromise } = require('internal/safe_globals');
+const resolvedPromise = SafePromise.resolve();
+const resolvedArrayPromise = SafePromise.resolve([]);
+const { ModuleWrap } = require('internal/loader/ModuleWrap');
+
+class ModuleJob {
+  /**
+   * @param {module: ModuleWrap?, compiled: Promise} moduleProvider
```

Comment:
We don't typically use such annotations, not that it is a bad thing though.

Reply:
going to wontfix since it doesn't sound like an issue.

Reply:
It's more of a note - I don't have much opinion on it but it is certainly not consistent with the rest of the codebase.